plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts

```diff
@@ -54,8 +55,21 @@ export class GkeClusterLocator implements KubernetesClustersSupplier {
         return { key: mrl.getString('key'), value: mrl.getString('value') };
       }) ?? [];
 
+    let storeAuthProviderString: string;
+    let getGkeProperty;
+    try {
+      getGkeProperty = config.getString('authProvider');
+    } catch (err) {
+      getGkeProperty = 'google';
+    }
+    if (getGkeProperty === 'googleServiceAccount') {
+      storeAuthProviderString = 'googleServiceAccount';
+    } else {
+      storeAuthProviderString = 'google';
+    }
```

Comment:
I might be refactoring too aggressively here:
```suggestion
    const storeAuthProviderString = config.getOptionalString('authProvider') === 'googleServiceAccount' ? 'googleServiceAccount' : 'google';
```

regardless, can we have some unit tests for this (they live [here](https://github.com/backstage/backstage/blob/master/plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.test.ts))? I think we'd want at least scenarios for:

* no `authProvider` specified
* `authProvider = 'googleServiceAccount'`
* `authProvider` is neither `googleServiceAccount` nor `google`

Furthermore, let's update the config schema here to mention this new parameter:

https://github.com/backstage/backstage/blob/2d59a19fd3a566dbb8f1639aab8c79f84db7db2a/plugins/kubernetes-backend/config.d.ts#L76-L87

Reply:
@jamieklassen  Better way to simplify!

Thank you!